backend/src/email/middlewares/email-transactional.middleware.ts

```diff
@@ -41,6 +45,27 @@ async function sendMessage(
   }
 }
 
+const rateLimit = expressRateLimit({
+  store: new RedisStore({
+    prefix: 'email-transactional:',
+    client: RedisService.rateLimitClient,
+    expiry: config.get('transactionalEmail.window'),
+  }),
+  keyGenerator: () => 'global',
```

Comment:
Just thinking, do you think it will make more sense to rate limit by user instead of imposing a system wide rate limit. 

Given the scenario with two senders, A and B. If sender A sends at a much faster rate than sender B, sender B will never have a chance to send.

Reply:
Good point. But I think a global rate limit is still a good thing to have alongside a user-level rate limit.

The user-level rate limit would prevent individual users from monopolizing the global rate limit. On the other hand, the global rate limit would prevent single-send emails as a whole from monopolizing the SES send rate.

What do you think?

Reply:
Yep. I think that makes sense.

Reply:
To implement this, I'll need to use a more flexible rate limiter instead, like `node-rate-limiter-flexible`. `express-rate-limit`'s API is not flexible enough.

Reply:
Alternatively, we can implement the global rate limit on CloudFlare. But that might make things confusing since we have rate limit at two different places. 

Another thought is that, given we have relatively small number of initial users for this. We can always ensure that `number of single send users` x `per user email rate limit` is less than the maximum number of concurrent single send we want to allow on Postman.

Reply:
I think the issue with having the global rate limit on CloudFlare is: if a sender gets past the global rate limit, but not the user rate limit, the global rate limit would still use a token even if it shouldn't.

This is the same issue with `express-rate-limit`. In general, I think that token consumption should be an all-or-nothing event -- either both the global and user rate limits use a token or, none of them do and the user is rate limited.